internal/host/static/repository_host_test.go

```diff
@@ -604,6 +606,12 @@ func TestRepository_UpdateHost(t *testing.T) {
 			assert.NoError(err)
 			require.NotNil(orig)
 
+			if tt.wantIsErr == 0 {
```

Comment:
I get an invalid memory address runtime error when I attempt to set `tt.want.SetIds` on line 612 when `tt.want` does not exist. I could wrap an if statement around that line only to check if `tt.want` exists. It is still the same number of if statements, but I think it might be slightly better for cognitive load. Do you have any alternate suggestions?

Reply:
The error will be due to the fact that `tt.want` is nil. An alternative would be to define the want in the test case instead of adding it during the test so something like:

```
want: &Host{
	Host: &store.Host{
		Name:        "test-update-name-repo",
		Description: "test-description-repo",
		Address:     "127.0.0.1",
	},
        SetIds: []string{....},
}
```
We would also want to make sure we handle all the test cases here, two off the top of my head:
- Orig does not have any sets and we add new sets in the update
- Orig has sets and we change them in the update path

This will likely require reworking this unit test a bit to create orig with and without sets outside of the loop here.

Reply:
@louisruch I'm not quite sure what you mean about the test cases. I thought `UpdateHost` only worked on the name, description, and address. 
I am currently running tests with a singular hostset to make sure the set is persisted, upon calling `UpdateHost`. I understand the meaning of the first bullet point, but if I were to test that, by the same logic, wouldn't I have to test every field that isn't name, description, or address?
For the second bullet point, could you expand what you mean by "change them in the update path"? Thanks

Reply:
Sorry you are right, we have a separate call for SetHostSets!